core/templates/dev/head/pages/exploration_editor/simple_editor_tab/ScrollSyncService.js

```diff
@@ -0,0 +1,126 @@
+// Copyright 2014 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service that syncs the position of the user in the simple editor
+ with the navigation sidebar.
+ */
+
+oppia.factory('ScrollSyncService', [
+  '$anchorScroll', '$document', '$rootScope', 'QuestionHashService',
+  'SimpleEditorManagerService',
+  function($anchorScroll, $document, $rootScope, QuestionHashService,
+           SimpleEditorManagerService) {
+    var targets = [];
+    var questionList = SimpleEditorManagerService.getQuestionList();
+    var clear = function() {
+      angular.element('.highlighted').removeClass('highlighted');
+    };
+    var collapse = function(hash) {
+      var question;
+      var questionIndex;
+      var questions = questionList.getQuestions();
+      for (var i = 0; i < questions.length; i++) {
+        if (QuestionHashService.getHash(questions[i]) === hash) {
+          questionIndex = i;
+          break;
+        }
+      };
+      if (!angular.isDefined(questionIndex)) {
+        return;
+      };
+      question = questions[questionIndex];
+      $rootScope.$broadcast('SimpleEditorSidebarLinkCollapse', question);
+    };
+    $anchorScroll.yOffset = 80;
+
+    var scrollSyncService = {
+      // Add an element to the targets list, when a user scrolls to an element
+      // in the target list, the sidebar is updated accordingly.
+      addTarget: function(target) {
+        targets.push(target);
+      },
+      // Update the active nav link
+      updateActive: function(hash) {
+        if (!angular.isDefined(hash)) {
+          return;
+        } else {
+          clear();
+          var elm = angular.element('#si-' + hash);
+
+          if (!(hash === 'intro' || hash === 'title')) {
+            elm.addClass('highlighted');
+          };
+
+          // Uncollapse the nav link in case hidden
+          if (elm.is(':hidden')) {
+            collapse(QuestionHashService.getParentQuestionHash(hash));
+          };
+          // TODO: scroll to the nav link in case not displayed
```

Comment:
TODO(andromfins): If the nav link is not displayed, scroll to it.

That said, I don't think I understand this todo. What nav link are you referring to, and under what circumstances is it not displayed? Some elaboration would be nice.